encode.go

```diff
@@ -400,20 +400,21 @@ func (e *Encoder) encodeByMarshaler(ctx context.Context, v reflect.Value, column
 	if t, ok := iface.(time.Time); ok {
 		return e.encodeTime(t, column), nil
 	}
+	// Handle *time.Time explicitly since it implements TextMarshaler and shouldn't be treated as plain text
+	if t, ok := iface.(*time.Time); ok && t != nil {
```

Comment:
This works correctly 👍 

Since `isInvalidValue` becomes true at the beginning of `encodeValue` when *time.Time is nil, this process is always executed only when *time.Time is not nil.